sdk/core/Azure.Core.Experimental/tests/DynamicData/DynamicJsonTests.cs

```diff
@@ -641,7 +641,7 @@ public void CanMixPascalCaseAndCamelCaseNames()
             Assert.AreEqual(1, (int)dynamicJson.Foo);
             Assert.AreEqual(3, (int)dynamicJson.bar);
             Assert.AreEqual(3, (int)dynamicJson.Bar);
-            Assert.AreEqual(null, dynamicJson["Bar"]);
+            Assert.Throws<KeyNotFoundException>(() => _ = dynamicJson["Bar"]);
```

Comment:
Does the dynamic casing option have any effect here, or is the indexer always strict pass-through?

Reply:
That's for us to choose (i.e. nothing in the implementation requires it to be a certain way), but I think we're converging on a "no magic in dictionary-style lookups" story that I like, so I would keep it a strict pass through without a strong reason to do otherwise.

Reply:
Makes sense - I was just curious because the test above seems to show case conversion with dynamic but not with the indexer. I assume that if we converge towards "no magic" by default, that the dynamic property lookups would be strict as well unless the casing options were provided. But I'm also wondering if the casing options on get should apply to indexers also.

Reply:
Yeah, we'll loop back on all these tests when we converge on the API and default behavior.  Right now, my sense is to keep indexers as a strict pass-through in all cases, but I'm open to compelling use cases that would drive a different decision.